spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/db/tasks/TaskExecutionQueryTests.java

```diff
@@ -0,0 +1,240 @@
+			long startTime = System.currentTimeMillis();
+			String[] tableNames = new String[] {
+				"BATCH_JOB_INSTANCE",
+				"BATCH_JOB_EXECUTION",
+				"BATCH_JOB_EXECUTION_CONTEXT",
+				"BATCH_JOB_EXECUTION_PARAMS",
+				"BATCH_STEP_EXECUTION",
+				"BATCH_STEP_EXECUTION_CONTEXT",
+				"TASK_EXECUTION",
+				"TASK_EXECUTION_METADATA",
+				"TASK_EXECUTION_PARAMS",
+				"TASK_TASK_BATCH"
+			};
+			Map<String, Map<String, Integer>> columnTypes = new HashMap<>();
+			add(columnTypes, "BATCH_JOB_INSTANCE", "VERSION", Types.INTEGER);
+			add(columnTypes, "BATCH_JOB_EXECUTION", "VERSION", Types.INTEGER);
+			add(columnTypes, "BATCH_STEP_EXECUTION", "VERSION", Types.INTEGER);
+			add(columnTypes, "BATCH_JOB_EXECUTION_PARAMS", "DATE_VAL", Types.TIMESTAMP);
+			add(columnTypes, "BATCH_JOB_EXECUTION_PARAMS", "LONG_VAL", Types.BIGINT);
+			add(columnTypes, "BATCH_JOB_EXECUTION_PARAMS", "DOUBLE_VAL", Types.DOUBLE);
+			add(columnTypes, "TASK_EXECUTION", "EXTERNAL_EXECUTION_ID", Types.VARCHAR);
+			add(columnTypes, "TASK_EXECUTION", "EXIT_CODE", Types.INTEGER);
+
+			for (String tableName : tableNames) {
+				long startLoad = System.currentTimeMillis();
+				logger.info("loading:{}", tableName);
+				Map<String, Integer> tableColumnTypes = columnTypes.get(tableName);
+				CSVLoader.DeriveType deriveType = columnName -> {
+					String col = columnName.toUpperCase();
+					Integer type = tableColumnTypes != null ? tableColumnTypes.get(col) : null;
+                    if(type != null) {
+                        return type;
+                    }
+                    if (col.equals("ID") || col.endsWith("_ID")) {
+                        return Types.BIGINT;
+                    }
+                    if (col.endsWith("_COUNT")) {
+                        return Types.INTEGER;
+                    }
+                    if (col.endsWith("_TIME") || col.endsWith("_UPDATED")) {
+                        return Types.TIMESTAMP;
+                    }
+                    return Types.VARCHAR;
+                };
+				ClassPathResource resource = new ClassPathResource("task-executions/" + tableName + ".csv");
+				assertThat(resource.exists()).withFailMessage(() -> "classpath:task-executions/" + tableName + ".csv:NOT FOUND").isTrue();
+				int loaded = CSVLoader.loadCSV(
+					tableName,
+					dataSource,
+					resource,
+					deriveType
+				);
+				long loadTime = System.currentTimeMillis() - startLoad;
+				logger.info("loaded:{} into {} in {}ms", loaded, tableName, loadTime);
+			}
+			long totalTime = System.currentTimeMillis() - startTime;
+			logger.info("Total load time={}ms", totalTime);
+		}
+
+		private void add(Map<String, Map<String, Integer>> columnTypes, String tableName, String columnName, int type) {
+			Map<String, Integer> tableColumnTypes = columnTypes.computeIfAbsent(tableName, k -> new HashMap<>());
+            tableColumnTypes.put(columnName, type);
+		}
+
+
+		@Test
+		void queryWithLargeNumberOfTaskExecutions() throws Exception {
```

Comment:
Not sure we should be testing speed in our Unit tests.  I think this is a test that is done to verify the change works, not one that is repeated for each release.

Reply:
The tests tasks a few seconds for each database and verifies a lot more

Reply:
I think all of these performance tests should be repurposed outside of the CI runs.

Reply:
Change to IT and created ci-performance.yml